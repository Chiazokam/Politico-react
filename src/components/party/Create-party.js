import React, {Component} from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { Container, FormField, Button } from '../global';
import { createParty, getParties } from '../../actions';
import '../../styles/create-party/create-party.scss';

class CreatePartyUnit extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      logoUrl: '',
      hqAddress: ''
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.setState({
      submit: true,
      errors: {},
    });
    const party = {
      name: this.state.name,
      logoUrl: this.state.logoUrl,
      hqAddress: this.state.hqAddress
    };
    const { parties, createParty } = this.props;
    createParty(party);
    parties();
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value});
  }

  clearField = (e) => {
    this.setState({ errors: '' })
  }

  render() {
    const { errors, redirect } = this.props;
    if (redirect) return <Redirect to='/parties'/>;
    
    return (
      <div className="create-party">
        <Container>
          <p className='form-text'>Create Party</p>
          <p className='form-info'>All fields are compulsory</p>
            <form onSubmit={this.handleSubmit.bind(this)}>
              <div className="error">{errors.message}</div>
              <FormField 
                className='form-field'
                type='text'
                error={this.props.errors.name}
                name='name'
                onFocus={this.clearField.bind(this)}
                onChange={this.handleChange.bind(this)}
                value={this.state.name}
                placeholder='Party Name'
              />

              <FormField 
                className='form-field'
                type='text'
                error={errors.logoUrl}
                name='logoUrl'
                onFocus={this.clearField.bind(this)}
                onChange={this.handleChange.bind(this)}
                value={this.state.logoUrl}
                placeholder='Logo Url'
              />

              <FormField 
                className='form-field'
                type='text'
                error={errors.hqAddress}
                name='hqAddress'
                onFocus={this.clearField.bind(this)}
                onChange={this.handleChange.bind(this)}
                value={this.state.hqAddress}
                placeholder='Party Head Quarters'
              />

              <Button
                type="submit"
                className="btn btn-colored btn-signin btn-dark">
                { this.state.submit && <FontAwesomeIcon 
                  icon={ faSpinner }
                  spin
                /> }
                &nbsp; Submit
              </Button>
            </form>
        </Container>
      </div>
    )
  }
}

const mapDispatchToProps = {
  parties: getParties,
  createParty,
};

const mapStateToProps = state => ({
  name: state.party.name,
  logoUrl: state.party.logoUrl,
  hqAddress: state.party.hqAddress,
  errors: state.party.errors,
  submit: state.party.submit,
  redirect: state.party.redirect
});

const CreateParty = connect(mapStateToProps, mapDispatchToProps)(CreatePartyUnit);

export { CreateParty, CreatePartyUnit };
