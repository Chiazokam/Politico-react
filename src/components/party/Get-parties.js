import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PartyCard } from '../global';
import { getParties } from '../../actions';

class GetParties extends Component {
  constructor(props){
    super(props);
  }
  componentDidMount() {
    this.getParties();
  }

  getParties = () => {
    const { dispatch } = this.props;
    const token = localStorage.getItem('token');
    const config = {
      headers: { token }
   };
    dispatch(getParties(config));
  }

  render() {
    const { getParties } = this.props;
    const isAdmin = localStorage.getItem('isAdmin');
    let editButton, deleteButton;
    if(isAdmin === 'true') {
      editButton =  <a className="edit">Edit Name</a>
      deleteButton = <a className="delete">Delete</a>
    } 
    else if (isAdmin === 'false') {
      editButton =  ''
      deleteButton = ''
    }
    
    return (
      <React.Fragment>
        {getParties.length === 0 ? 
        <div>
          <div className="card-party title-header">
            <div className="party-main-admin party-main-parties admin-view-main">No Parties Found</div>
          </div>
        </div> :
        <div>
          <div className="card-party title-header">
            <div className="party-main">Parties</div>
            <div className="party-logo">Logo</div>
          </div>
          <div className="card-party title-mobile">
            <div className="party-main">Parties</div>
          </div>
        </div> }

        { getParties.map(party => 
          <PartyCard
            key={Math.random(Date.now())}
            name={party.name}
            headquarters={party.hqaddress}
            logourl={party.logourl}
            edit={editButton}
            delete={deleteButton}
        />
        )}
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => ({
  getParties: state.party.getParties,
});

export default connect(mapStateToProps)(GetParties);
