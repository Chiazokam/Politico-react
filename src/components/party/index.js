import { CreateParty } from './Create-party';
import { GetParties }from './Get-parties';

export { CreateParty, GetParties };
