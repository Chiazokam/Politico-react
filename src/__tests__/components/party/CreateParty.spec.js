import React from 'react';
import "@babel/polyfill";
import { shallow } from 'enzyme';
import { CreatePartyUnit } from '../../../components/party/Create-party';

describe('<Create Party />', () => {
  it('renders correctly', () => {
    const props = {
      errors: {
        message: 'Wrong input',
        party: 'New Party',
      },
      getParties: [],
      parties: jest.fn()
    };
    const tree = shallow(<CreatePartyUnit {...props} />);
    expect(tree).toMatchSnapshot(); 
  });
});
