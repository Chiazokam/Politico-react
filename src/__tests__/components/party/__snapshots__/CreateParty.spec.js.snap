// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create Party /> renders correctly 1`] = `ShallowWrapper {}`;
